src/Providers/FeatureLayer.js

```diff
@@ -95,7 +95,11 @@ export var FeatureLayerProvider = FeatureLayerService.extend({
       queryString.push(field + " LIKE upper('%" + text + "%')");
     }
 
-    return queryString.join(' OR ');
+    if (this.options.where) {
+      return this.options.where + ' AND ' + queryString.join(' OR ');
```

Comment:
The issue raises with more than one search field.
The current expression is something like:
IS_ENABLED=1 AND upper(NAME) like ... OR upper(SURNAME) like ... 
The one I'd propose is
IS_ENABLED=1 AND (upper(NAME) like ... OR upper(SURNAME) like ... )
The parentheses guarantee  the right evaluation order.